src/logic/safe/store/actions/fetchSafe.ts

```diff
@@ -1,178 +1,187 @@
-      abi: GnosisSafeSol.abi as AbiItem[],
-      address: safeAddress,
-      methods: safeParams,
-    }),
+export const buildSafe = async (safeAddress: string, safeName: string): Promise<SafeRecordProps> => {
+  const checksummedSafeAddress = checksumAddress(safeAddress)
+  const safeInfo: Partial<SafeRecordProps> = {
+    address: checksummedSafeAddress,
+    name: safeName,
+  }
+
+  const [remoteSafeInfoSettled, localSafeInfoSettled, safeBalancesSettled] = await Promise.allSettled([
     getSafeInfo(safeAddress),
     getLocalSafe(safeAddress),
+    fetchTokenCurrenciesBalances({ safeAddress, selectedCurrency: 'USD' }),
   ])
 
-  // request SpendingLimit info
-  const spendingLimits = safeInfo ? await getSpendingLimits(safeInfo.modules, safeAddress) : null
-
-  // Converts from [ { address, ownerName} ] to address array
-  const localOwners = localSafe ? localSafe.owners.map((localOwner) => localOwner.address) : []
+  // remote (client-gateway)
+  const remoteSafeInfo = await extractRemoteSafeInfo(remoteSafeInfoSettled, checksummedSafeAddress)
 
-  const modules = await getModules(safeInfo)
+  // local (localStorage)
+  const localSafeInfo = await extractLocalSafeInfo(localSafeInfoSettled, remoteSafeInfoSettled.status === 'fulfilled')
 
-  const updatedSafe = {
-    address: safeAddress,
-    name: localSafe?.name,
-    modules,
-    spendingLimits,
-    nonce: remoteNonce ? Number(remoteNonce) : undefined,
-    threshold: remoteThreshold ? Number(remoteThreshold) : undefined,
-    featuresEnabled: localSafe?.currentVersion ? enabledFeatures(localSafe.currentVersion) : localSafe?.featuresEnabled,
-  }
+  // update owner's information
+  const owners = extractSafeOwners(remoteSafeInfoSettled, localSafeInfoSettled)
 
-  dispatch(updateSafe(updatedSafe))
+  // remote (client-gateway)
+  // TODO: REVIEW: do we really need to load ethBalance?
+  //  it's being loaded by the `fetchSafeTokens` action anyway
+  safeInfo.ethBalance = extractSafeNativeBalance(safeBalancesSettled)
 
-  if (!remoteOwners.length) {
-    return
-  }
-  // If the remote owners does not contain a local address, we remove that local owner
-  localOwners.forEach((localAddress) => {
-    const remoteOwnerIndex = remoteOwners.findIndex((remoteAddress) => sameAddress(remoteAddress, localAddress))
-    if (remoteOwnerIndex === -1) {
-      dispatch(removeSafeOwner({ safeAddress, ownerAddress: localAddress }))
-    }
-  })
-
-  // If the remote has an owner that we don't have locally, we add it
-  remoteOwners.forEach((remoteAddress) => {
-    const localOwnerIndex = localOwners.findIndex((localAddress) => sameAddress(remoteAddress, localAddress))
-    if (localOwnerIndex === -1) {
-      dispatch(
-        addSafeOwner({
-          safeAddress,
-          ownerAddress: remoteAddress,
-          ownerName: 'UNKNOWN',
-        }),
-      )
-    }
-  })
+  return { ...safeInfo, ...remoteSafeInfo, ...localSafeInfo, owners } as SafeRecordProps
```

Comment:
Why is `localSafeInfo` going after the `remoteSafeInfo`?
Shouldn't the backend be the source of truth for the initial load?

Reply:
Maybe the approach of changing the spread order can be more interesting. I was also having troubles while checking the local values info function. I don't think that avoid adding the values to localSafeInfo save us load effort and could lead to a more simple code. :point_up:

Reply:
along with other changes, the spread was modified, which simplified how local safe info is recovered, which lead to simply delete the `extractLocalSafeInfo` helper function.

Actually, the only place where `localSafeInfo` is used, is in `buildSafe` function (we may be able to load the information from the redux store, but that's something to work on the "load safe flow improvement")